R/compile_model.R

```diff
@@ -17,18 +17,17 @@
 ##' Function to compile custom \code{\link{SimInf_model}}.
```

Comment:
I would like to keep the `SimInf_model` class unchanged and instead define a new class (that contains a `SimInf_model`) to handle the compiled model. It's then possible to create a run function to handle the new class.   

```
##' Class to handle a compiled custom \code{\link{SimInf_model}}
##'
##' @section Slots:
##' \describe{
##'   \item{filename}{
##'     Character vector of length 1 containing name of shared library.
##'     This is used internally to deal with models that have been
##'     compiled using \code{compile_model}.
##'   }
##'   \item{filename}{
##'     A character vector of length 1 specifying the name of the
##'     custom model.
##'   }
##' }
##' @include SimInf_model.R
##' @export
setClass("SimInf_model_dll",
         contains = c("SimInf_model"),
         slots = c(filename = "character",
                   name     = "character"))

##' @rdname run
##' @export
setMethod("run",
          signature(model = "SimInf_model_dll"),
          function(model, threads, solver)
          {
              solver <- match.arg(solver)

              ## Check that SimInf_model contains all data structures
              ## required by the siminf solver and that they make sense
              validObject(model);

              dll <- dyn.load(model@filename)
              on.exit(dyn.unload(model@filename), add = TRUE)

              ## Create expression to parse
              expr <- ".Call(dll$SimInf_model_run, model, threads, solver)"

              ## Run the model. Re-throw any error without the call
              ## included in the error message to make it cleaner.
              tryCatch(eval(parse(text = expr)), error = function(e) {
                  stop(e$message, call. = FALSE)
              })
          }
)
```

Reply:
Great idea. Thanks Stefan. Let me implement those changes and update the pull request.